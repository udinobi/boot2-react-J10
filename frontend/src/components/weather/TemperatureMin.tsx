import React from "react"

import styled from "styled-components"

import { SmallerText } from "../lib/StyledText"

export interface Data {
    minTemperature: number
}

const Span = styled.span`
    font-family: "Roboto", "sans-serif";
`

const MinTemperature = styled(SmallerText)`
    display: inline;
`

const Icon = styled.i`
    color: #0093DC;
    font-size: 1.4rem;
    padding-right: 10px;
    padding-left: 3px;
`

export default (props: Data) => 
    <div style={{ marginTop: 6 }}>
        <Icon className="wi wi-thermometer-exterior" />
        <MinTemperature>
            <Span>{Math.round(props.minTemperature)}</Span>
            <sup>Â°C</sup> {"\u00A0"}(min.)
        </MinTemperature>
    </div>
