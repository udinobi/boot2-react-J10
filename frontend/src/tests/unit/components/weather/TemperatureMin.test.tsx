import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import React from "react"

import TemperatureMin, { Data } from "../../../../components/weather/TemperatureMin"

describe("--> Snapshot for Components > weather > 'TemperatureMin'", () => {
    let props: Data

    beforeEach(() => {
        props = {
            minTemperature: 0
        }
    })

    it("matches", () => {
        const wrapper = shallow(<TemperatureMin {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
