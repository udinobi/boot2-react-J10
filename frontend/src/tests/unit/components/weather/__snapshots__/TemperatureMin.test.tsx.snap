// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--> Snapshot for Components > weather > 'TemperatureMin' matches 1`] = `
<div
  style={
    Object {
      "marginTop": 6,
    }
  }
>
  <styled.i
    className="wi wi-thermometer-exterior"
  />
  <Styled(Styled(Styled(styled.div)))>
    <styled.span>
      0
    </styled.span>
    <sup>
      °C
    </sup>
     
     
    (min.)
  </Styled(Styled(Styled(styled.div)))>
</div>
`;
